src/torque-slides.jsx

```diff
@@ -32,6 +40,23 @@ var TorqueSlides = React.createClass({
     this.setInterval(this.updateSlide, 1000);
   },
 
+  keydown(e) {
+    switch (e.keyCode) {
+      case RIGHT_ARROW:
+        this.setState({
+          current: (this.state.current + 1) % this.props.children.length,
+          countdown: this.props.duration
```

Comment:
Also the onVisible callback. You might want to pull out a lot of the logic of update slide into a separate function and then call it in keydown and update slide.